Ruins/src/main/java/atomicstryker/ruins/common/RuinTemplate.java

```diff
@@ -39,14 +40,18 @@
     private boolean preventRotation = false;
     private final ArrayList<Integer> bonemealMarkers;
     private final ArrayList<AdjoiningTemplateData> adjoiningTemplates;
+    
+    private static Hashtable<String,RuinTemplate> knownTemplates = new Hashtable<>();

```

Comment:
Instead of this, why not refactor FileHandler.templates into something static to get access to instances? Have the Ruin file Loader run once over all templates without actually instantiating the adjoining templates, then "connect" them by reference. You are essentially keeping a second set of RuinTemplate objects here.